proxy/threads.c

```diff
@@ -0,0 +1,176 @@
+#define _GNU_SOURCE
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <memory.h>
+#include <pthread.h>
+
+
+#include "queue.h"
+#include "sockets.h"
+#include "mypoll.h"
+#include "threads.h"
+
+#define BUFFER_SIZE 4096
+
+
+
+void* connectionThread(void* arg) {
+    Data d = *(Data*)arg;
+
+    queue_t* newConnectQueue = d.queue;
+    sPoll* spoll = d.poll;
+    //ssize_t bytes_read;
+
+    for (;;) {
+        Sockets* newSockets;
+        queue_get(newConnectQueue, (void**)&newSockets);
+        char buffer[BUFFER_SIZE];
+        // Чтение запроса от клиента
+        int readed;
+        //printf("начал читать %d\n", newSockets->client);
+        if (( readed = read(newSockets->client, buffer, BUFFER_SIZE)) > 0) {
+        }
+
+        if (readed == -1) {
+            printf("соединение разорвано\n");
+            closeSockets(*newSockets);
+            free(newSockets);
+            continue;
+        }
+
+        int dnslen;
+        char *start, *end;
+
+        start = strstr(buffer, "//");
```

Comment:
with knowledge of your code here I can break your proxy and crach your proxy., Explain how and fix it.

Reply:
?